packages/react-dom/src/__tests__/ReactDOMSelect-test.js

```diff
@@ -557,6 +557,34 @@ describe('ReactDOMSelect', () => {
     }
   });
 
+  it('should warn if selected is set on <option>', () => {
+    spyOnDev(console, 'error');
+
+    ReactTestUtils.renderIntoDocument(
+      <select>
+        <option selected={true} />
+        <option selected={true} />
+      </select>,
+    );
+    if (__DEV__) {
+      // Test deduplication
```

Comment:
This comment is not relevant here because you’re *not* testing deduplication :-)
To actually test deduplication, move it *after* the second render.

Reply:
@gaearon  Just so I understand. I am doing the following twice which should fire the warning twice:

```html
<option selected={true} />
<option selected={true} />
```

And so isn't this testing for deduplication of the warning?

Reply:
It depends on how deduplication is implemented—in general I'd say rendering twice is a more sure way to tell if it works or not. You're right that in this case it's enough, but I would feel safer if we render twice and then check.

Reply:
Sounds good